### PR TITLE
Added missing properties to Milestone Class

### PR DESCRIPTION
**Pull request by shaggygi**
This is related to issue #1358.

Added the following properties: HtmlUrl, LabelsUrl, Id, UpdatedAt
Added constructor with all properties (except Id).
Rearranged the properties to match the order from JSON response.
Added a few comment changes.

### CODE REVIEW

**Comment by shaggygi** [comment]
This is related to issue #1358.

**Comment by shiftkey** [comment]
@shaggygi it'd be nice to add an integration test to [`MilestonesClientTests`](https://github.com/octokit/octokit.net/blob/master/Octokit.Tests.Integration/Clients/MilestonesClientTests.cs) to verify these new properties are deserialized correctly, but aside from that this is looking good!

**Comment by shaggygi** [comment]
@shiftkey Thanks for feedback.  I was thinking I should have added tests based on the contribution notes, but didn't know exactly where being new to this repository.  I will review what you mentioned and try to add soon.

**Comment by shiftkey** [comment]
@shaggygi lots of the existing tests are written in the "arrange-act-assert" pattern, so feel free to steal from them if you're looking for guidance...

**Comment by shaggygi** [comment]
I added a test using [BlobClientTests](https://github.com/octokit/octokit.net/blob/f354d1bf00ff7606b46489e3a915e1428414bc47/Octokit.Tests.Integration/Clients/BlobClientTests.cs) as example.  Did not verify Creator property, but left a TODO comment.  If you know of a similar test to verify User, let me know and I can update.

**Comment by ryangribble** [comment]
so what you've got there is actually a unit test, that is providing a hardcoded json framgent and testing the deserialisation logic only.  

This is great, but what @shiftkey was referring to was adding an integration test (a test that actually runs against github.com) to [the MilestoneClientTests ](https://github.com/octokit/octokit.net/blob/master/Octokit.Tests.Integration/Clients/MilestonesClientTests.cs).  There is already a test there [CanRetrieveOneMilestone](https://github.com/octokit/octokit.net/blob/master/Octokit.Tests.Integration/Clients/MilestonesClientTests.cs#L26) that you could just add some more asserts to, to prove that your new fields are being returned correctly.

eg:
``` csharp
// existing assert
Assert.Equal("a milestone", result.Title); 
// Add your asserts:
Assert.Equal("xxxx", result.HtmlUrl);
Assert.Equal("xxxx", result.LabelsUrl);
etc
```

**Comment by ryangribble** [comment]
As general comments on this change, I'm :+1: for including useful fields like `HtmlUrl` `UpdatedAt` and even `Id` for a milestone (and various other objects/models)... 

but I'm not sure about things like `LabelsUrl`.  Simply because those fields are API hypermedia links (eg in this case, the API url to get the labels for this milestone).  Since octokit already provides that information via interfaecs and clients (eg the `IssuesLabelsClient.GetAllForMilestone()` call), Im not sure we need to include this field on the `Milestone` object itself.  Or more to the point, if we do include `LabelsUrl` then we should include all the other `xxxUrl` fields (like events, starred, followers, following, gists, etc).  

@shiftkey might have an opinion on this, but my 2c would be to remove `LabelsUrl` from this change...

**Comment by shaggygi** [comment]
After reviewing/learning more about the API structure, I think I agree with @ryangribble on removing LabelsUrl.  It is nice to have the property included, but logic is already in place for IssuesLabelsClient.  I will remove and commit soon.  I originally added similar to others since it was returned in the JSON.

File: Octokit.Tests.Integration/Clients/MilestonesClientTests.cs [hunk 1]
```diff
@@ -25,12 +25,33 @@ public MilestonesClientTests()
...
 
-        Assert.Equal("a milestone", result.Title);
+        var url = "https://api.github.com/repos/" + _context.RepositoryOwner + "/" + _context.RepositoryName + "/milestones/" + created.Number;
+        Assert.Equal(new Uri(url), result.Url);
+        var htmlUrl = "https://github.com/" + _context.RepositoryOwner + "/" + _context.RepositoryName + "/milestones/" + result.Title;
+        Assert.Equal(new Uri(htmlUrl), result.HtmlUrl);
+        Assert.True(result.Id > 0);
+        Assert.Equal(title, result.Title);
+        Assert.Equal(description, result.Description);
+        // TODO: Test Creator
```
Comment (ryangribble) [inline Octokit.Tests.Integration/Clients/MilestonesClientTests.cs]: To test for creators fields (it's just a `User` object) you should just be able to use similar to the following:

``` csharp
Assert.Equal(Helper.UserName, result.Creator.Login);
```

File: Octokit.Tests/Clients/MilestonesClientTests.cs [hunk 2]
```diff
@@ -230,5 +232,71 @@ public void EnsuresNonNullArguments()
...
+
+            Assert.NotNull(response.Body);
+            Assert.Equal(milestoneResponseJson, (string)response.HttpResponse.Body);
+            Assert.Equal(new Uri("https://api.github.com/repos/octokit/octokit.net/milestones/6"), response.Body.Url);
+            Assert.Equal(new Uri("https://github.com/octokit/octokit.net/milestones/Pagination%20Support"), response.Body.HtmlUrl);
+            Assert.Equal(1641780, response.Body.Id);
+            Assert.Equal(6, response.Body.Number);
+            Assert.Equal("Pagination Support", response.Body.Title);
+            Assert.Equal("This milestone tracks deploying pagination out across the entire API", response.Body.Description);
+            // TODO: Test Creator
```
Comment (ryangribble) [inline Octokit.Tests/Clients/MilestonesClientTests.cs]: To test for creators fields (it's just a `User` object) you should just be able to use similar to the following:

``` csharp
Assert.Equal("shiftkey", result.Creator.Login);
Assert.Equal(1641780, result.Creator.Id);
```
etc

**Comment by ryangribble** [comment]
Ive just commented on how to tet the `Creator` property in your tests.

Other than that, you just need to update latest master into your branch and we should be good!

**Comment by shaggygi** [comment]
@ryangribble well, i've completely screw this up.  your suggestions worked on the user for test.  something when weird when merging and last file changes are not showing up.  many reverts/tries on my branch trying to understand more about VS/Git.  any thoughts on steps to address... it is clear one of my first few times on this :frowning:

**Comment by ryangribble** [comment]
So looking at your fork of the repo, it seems you merged your change into the master branch in YOUR fork, and wherever you sent this PR from doesnt actually exist anymore (did you delete your fork and re-fork it or something?  Or just delete the branch it used to be on perhaps?).  Anyway, I think you're going to need to close this PR and recreate it, once youve moved your work to a suitable branch.  At least the work you did is not lost though!  :grinning: 

Im not sure how much you've already read or not, so dont be offended if Im telling you something you already know, but as a contributor you have a fork of this repo under your own user space (this is referred to as `origin` and then you have this main repository (referred to as `upstream`).  Your local clone on your system will already have an `origin` remote pointing at your fork, and you should add an `upstream` remote pointing here (if you haven't already).

You should never make any changes in your master branch, instead you need to [Sync your fork with upstream](https://help.github.com/articles/syncing-a-fork/) to keep your master inline with master here.  

To do any work, you cut a branch from latest master, make your changes on that branch.  Always keep that branch up to date with master here (eg if we merge another PR before yours, you should pull master into your pending branch and your PR will update accordingly).  Eventually you send a PR from your fork/branch to the master branch here.  **Only when that PR is merged** should you then sync your master with upstream master (ie you should never push your changes into master on your side yourself, your master should only ever be synched with `upstream/master`

Im not sure about Visual Studio git GUI specifically, it may make it possible for you to just deal with `upstream/master` directly, and not have to really worry about `origin/master` much at all.  When teaching people git I tend to encourage consistent initial use of the basic command line commands, to gain the underlying understanding of what is actually happening.  Then we move to GUIs - a religious and contentious space :)  My personal preference is SourceTree but there are many options - GitHub Desktop, Git Extensions, and even VS2015 as you've mentioned (just make sure you're on the very latest Update and also add the GitHub Extension for some nice sugar when dealing with GitHub).


Anyway, what we need to do to fix you up is:

- Add your upstream remote if you dont already have it
- Move your changes to a branch and push it to your fork
- Reset your `origin/master` to `upstream/master` and force push it to your fork
- Ensure your branch is up to date with `upstream/master`
- And then of course raise a new PR for this milestone work

Then you should be :gem:


Off the top of my head, these would be the commands to do this, forgive any typos/syntax errors :grinning:

- Add your upstream remote if you dont already have it
```
git remote add upstream https://github.com/octokit/octokit.net.git
```

- Move your changes to a branch and push it to your fork
```
git checkout -b milestones-properties
# this next command just ensures you are up to date with your own master
git pull origin master
git push origin milestones-properties
```

- Reset your `origin/master` to `upstream/master`
```
git checkout master
git reset --hard upstream/master
git push -f origin master
```

- Ensure your branch is up to date with `upstream/master`
```
git checkout milestones-properties
git pull upstream master
# fix any merge issues etc
git push origin milestones-properties
```

Then you can raise the new PR

Hope that helps?

**Comment by shaggygi** [comment]
@ryangribble I really appreciate the info.  Like you mentioned, I have the changes so I can add later.  I'm going to go ahead and close the PR and review your notes (and more about Git terminology).  I'll circle back soon and attempt another PR when ready.

**Pull request closed without merging**